jbmc/unit/java_bytecode/java_bytecode_parser/parse_java_annotations.cpp

```diff
@@ -0,0 +1,90 @@
+/*******************************************************************\
+
+ Module: Unit tests for converting annotations
+
+ Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include <java-testing-utils/load_java_class.h>
+#include <testing-utils/catch.hpp>
+#include <java_bytecode/java_types.h>
+#include <java_bytecode/java_bytecode_parse_tree.h>
+#include <java_bytecode/java_bytecode_convert_class.h>
+
+// See
+// https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.7.16.1
+SCENARIO(
+  "java_bytecode_parse_annotations",
+  "[core][java_bytecode][java_bytecode_parser]")
+{
+  GIVEN("Some class files in the class path")
+  {
+    WHEN("Parsing an annotation with Class value specified to non-primitive")
+    {
+      const symbol_tablet &new_symbol_table = load_java_class(
+        "ClassWithClassTypeAnnotation",
+        "./java_bytecode/java_bytecode_parser");
+
+      THEN("The annotation should store the correct type")
+      {
+        const symbolt &class_symbol =
+          *new_symbol_table.lookup("java::ClassWithClassTypeAnnotation");
+        const std::vector<java_annotationt> &java_annotations =
+          to_annotated_type(class_symbol.type).get_annotations();
+        java_bytecode_parse_treet::annotationst annotations;
+        convert_java_annotations(java_annotations, annotations);
+        REQUIRE(annotations.size() == 1);
+        const auto &annotation = annotations.front();
+        const auto &element_value_pair = annotation.element_value_pairs.front();
+        const auto &id = to_symbol_expr(element_value_pair.value)
+          .get_identifier();
+        REQUIRE(id2string(id) == "Ljava/lang/String;");
+      }
+    }
+    WHEN("Parsing an annotation with Class value specified to primitive")
+    {
+      const symbol_tablet &new_symbol_table = load_java_class(
+        "ClassWithPrimitiveTypeAnnotation",
+        "./java_bytecode/java_bytecode_parser");
+
+      THEN("The annotation should store the correct type")
+      {
+        const symbolt &class_symbol =
+          *new_symbol_table.lookup("java::ClassWithPrimitiveTypeAnnotation");
+        const std::vector<java_annotationt> &java_annotations =
+          to_annotated_type(class_symbol.type).get_annotations();
+        java_bytecode_parse_treet::annotationst annotations;
+        convert_java_annotations(java_annotations, annotations);
+        REQUIRE(annotations.size() == 1);
+        const auto &annotation = annotations.front();
+        const auto &element_value_pair = annotation.element_value_pairs.front();
+        const auto &id = to_symbol_expr(element_value_pair.value)
+          .get_identifier();
+        REQUIRE(id2string(id) == "B");
```

Comment:
as above, this should probably be a CPROVER interanal java type and not the raw JVM type

Reply:
in `java_types.h` we have
```
typet java_int_type();
typet java_long_type();
typet java_short_type();
typet java_byte_type();
typet java_char_type();
typet java_float_type();
typet java_double_type();
typet java_boolean_type();
```

Reply:
Now calling `java_type_to_string` and comparing it with `java_byte_type()`.